include/flatbuffers/flatbuffers.h

```diff
@@ -515,6 +509,25 @@ class DetachedBuffer {
   size_t reserved_;
   uint8_t *cur_;
   size_t size_;
+
+  inline void destroy() {
+    if (buf_) {
+      assert(allocator_);
+      allocator_->deallocate(buf_, reserved_);
+    }
+    if (own_allocator_ && allocator_) {
+      delete allocator_;
+    }
```

Comment:
Since this may now be run outside of the dtor, I would assign nullptrs to guarantee that stuff isn't double freed.